templates/frontend/components/navigationMenu.tpl

```diff
@@ -23,17 +23,17 @@
 				<a href="{$navigationMenuItemAssignment->navigationMenuItem->getUrl()}">
 					{$navigationMenuItemAssignment->navigationMenuItem->getLocalizedTitle()}
 				</a>
-				{if !empty($navigationMenuItemAssignment->children)}
-					<ul>
-						{foreach key=childField item=childNavigationMenuItemAssignment from=$navigationMenuItemAssignment->children}
-							{if $childNavigationMenuItemAssignment->navigationMenuItem->getIsDisplayed()}
-								<li class="{$liClass|escape}">
-									<a href="{$childNavigationMenuItemAssignment->navigationMenuItem->getUrl()}">
-										{$childNavigationMenuItemAssignment->navigationMenuItem->getLocalizedTitle()}
-									</a>
-								</li>
-							{/if}
-						{/foreach}
+				{if $navigationMenuItemAssignment->navigationMenuItem->getIsChildVisible()}
+                    <ul>
+					{foreach key=childField item=childNavigationMenuItemAssignment from=$navigationMenuItemAssignment->children}
+						{if $childNavigationMenuItemAssignment->navigationMenuItem->getIsDisplayed()}
+							<li class="{$liClass|escape}">
+								<a href="{$childNavigationMenuItemAssignment->navigationMenuItem->getUrl()}">
+									{$childNavigationMenuItemAssignment->navigationMenuItem->getLocalizedTitle()}
+								</a>
+							</li>
+						{/if}
+					{/foreach}
```

Comment:
Some indentation issues with the `{foreach}` and the `<ul>` above.

Reply:
Fixed